### PR TITLE
proposed values are opened when clicking in the input textbox

### PR DESCRIPTION
**Pull request by florianbaer**
Implement open of dropdown when clicking into the input textbox

### CODE REVIEW

File: src/autocomplete.js [hunk 1]
```diff
@@ -312,6 +318,7 @@
             return data.filter(x => ~x[this.options.nameProperty].toLowerCase().indexOf(input.toLowerCase()));
         },
         filterOn: 'input',
+        onFocus: 'click',
```
Comment (philippdrebes) [inline src/autocomplete.js]: Rename this property to "openOnInput" which takes a boolean.

File: src/autocomplete.js [hunk 2]
```diff
@@ -119,13 +123,15 @@
         }
 
         initializeEventHandlers() {
+            this.$input[0].addEventListener(this.options.onFocus, this._onFocusHandler);
```
Comment (philippdrebes) [inline src/autocomplete.js]: If "openOnInput" is true, add handler like this:
`this.$input[0].addEventListener('input', this._onInputHandler);`

File: src/autocomplete.js [hunk 3]
```diff
@@ -50,6 +50,10 @@
                 }
             }.bind(this.$input[0]);
 
+            this._onFocusHandler = function () {
```
Comment (philippdrebes) [inline src/autocomplete.js]: `this.__onInputHandler = function () {`

**Review by philippdrebes — COMMENTED** [review]
I propose to achieve the desired behavior by implementinng it like this...

File: src/autocomplete.js (hunk 1 above)
Comment (florianbaer) [reply src/autocomplete.js]: Would you like to have the possibility to set this with HTML attributes: e.g. ignore-open-on-input?

Comment (philippdrebes) [reply src/autocomplete.js]: Sure why not. It's not a requirement, but nice to have. Though I think `open-on-input` would be a better name for the attribute (same as the property).

File: test/autocomplete.spec.js [hunk 4]
```diff
@@ -195,6 +195,48 @@ describe('Autocomplete', function () {
...
+
+        // fire the input event as if someone was typing
+        fireInputEvent($input[0]);
+
+        expect(dropdown).not.toBeEmpty();
+        $('body').click();
+        expect(dropdown).toBeEmpty()
+    });
+
+    it('should open the menu if clicked in the textbox', function () {
```
Comment (philippdrebes) [inline test/autocomplete.spec.js]: description does not match the test

File: test/autocomplete.spec.js [hunk 5]
```diff
@@ -195,6 +195,48 @@ describe('Autocomplete', function () {
...
+        expect(dropdown).toBeEmpty();
+
+        $input.val('a');
+
+        expect($input.val()).toBe('a')
+
+        // fire the input event as if someone was typing
+        fireInputEvent($input[0]);
+
+        expect(dropdown).not.toBeEmpty();
```
Comment (philippdrebes) [inline test/autocomplete.spec.js]: Does this really test if the dropdown is open?
Shouldn't it test if the dropdown is visible? Afaik `expect(dropdown).not.toBeEmpty();` only tests if there are elements in the DOM and not if they are visible.

Comment (florianbaer) [reply test/autocomplete.spec.js]: You are right sire :shipit:

**Review by philippdrebes — APPROVED** [review]

**Pull request merged**